salt/utils/boto.py

```diff
@@ -179,8 +179,10 @@ def get_connection(service, module=None, region=None, key=None, keyid=None,
     '''
 
     module = module or service
+    module = 'boto.' + module
+    module, submodule = module.rsplit('.', 1)
```

Comment:
Ah. Well OK, but in this case if `module` contains sub-submodule like `module.submodule.subsubmodule` you have a problem. Additionally, concatenating strings is not the very best style. I would still write it this way (oneliners rules, LOL):
```python
module, submodule = (lambda mod ['.'.join(pk) for pk in [mod[:2], mod[2:]]])(['boto'] + module.split('.'))
```